# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/mtransport/nr_socket_proxy.cpp
# Commit: c684f79d8c67
# Full Hash: c684f79d8c673f21179fce00358d70845a9692c0
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-09-18 21:49:27
# Regressor Bug: 1569183
# File Overlap Count: 2
# Description:
#   Bug 1569183: Rename PWebrtcProxyChannel to PWebrtcTCPSocket. r=mjf
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D45098
# ==============================================================================

diff -r 82af7b115f60 -r c684f79d8c67 media/mtransport/nr_socket_proxy.cpp
--- a/media/mtransport/nr_socket_proxy.cpp	Tue Sep 17 18:15:23 2019 +0000
+++ b/media/mtransport/nr_socket_proxy.cpp	Tue Sep 17 18:15:41 2019 +0000
@@ -43,7 +43,7 @@
 
 #include "mozilla/ErrorNames.h"
 
-#include "WebrtcProxyChannelWrapper.h"
+#include "WebrtcTCPSocketWrapper.h"
 
 namespace mozilla {
 using namespace net;
@@ -68,14 +68,14 @@
     : mClosed(false),
       mReadOffset(0),
       mConfig(aConfig),
-      mWebrtcProxyChannel(nullptr) {
+      mWebrtcTCPSocket(nullptr) {
   r_log(LOG_GENERIC, LOG_DEBUG, "NrSocketProxy::NrSocketProxy %p\n", this);
   MOZ_ASSERT(mConfig, "config should not be null");
 }
 
 NrSocketProxy::~NrSocketProxy() {
   r_log(LOG_GENERIC, LOG_DEBUG, "NrSocketProxy::~NrSocketProxy %p\n", this);
-  MOZ_ASSERT(!mWebrtcProxyChannel, "webrtc proxy channel not null");
+  MOZ_ASSERT(!mWebrtcTCPSocket, "webrtc proxy channel not null");
 }
 
 int NrSocketProxy::create(nr_transport_addr* aAddr) {
@@ -104,9 +104,9 @@
     return R_FAILED;
   }
 
-  mWebrtcProxyChannel = new WebrtcProxyChannelWrapper(this);
+  mWebrtcTCPSocket = new WebrtcTCPSocketWrapper(this);
 
-  mWebrtcProxyChannel->AsyncOpen(host, port, mConfig);
+  mWebrtcTCPSocket->AsyncOpen(host, port, mConfig);
 
   // trigger nr_socket_buffered to set write/read callback
   return R_WOULDBLOCK;
@@ -122,9 +122,9 @@
   mClosed = true;
 
   // We're not always open at this point.
-  if (mWebrtcProxyChannel) {
-    mWebrtcProxyChannel->Close();
-    mWebrtcProxyChannel = nullptr;
+  if (mWebrtcTCPSocket) {
+    mWebrtcTCPSocket->Close();
+    mWebrtcTCPSocket = nullptr;
   }
 }
 
@@ -147,7 +147,7 @@
     writeData.SetLength(aCount);
     memcpy(writeData.Elements(), aBuffer, aCount);
 
-    mWebrtcProxyChannel->SendWrite(std::move(writeData));
+    mWebrtcTCPSocket->SendWrite(std::move(writeData));
   }
 
   return 0;
@@ -218,7 +218,7 @@
   return R_INTERNAL;
 }
 
-// WebrtcProxyChannelCallback
+// WebrtcTCPSocketCallback
 void NrSocketProxy::OnClose(nsresult aReason) {
   nsCString errorName;
   GetErrorName(aReason, errorName);
@@ -260,7 +260,7 @@
 
   // We're always ready to write after we're connected. The parent process will
   // buffer writes for us.
-  if (!mClosed && mWebrtcProxyChannel && (poll_flags() & PR_POLL_WRITE) != 0) {
+  if (!mClosed && mWebrtcTCPSocket && (poll_flags() & PR_POLL_WRITE) != 0) {
     fire_callback(NR_ASYNC_WAIT_WRITE);
   }
 }
@@ -279,9 +279,8 @@
   return count;
 }
 
-void NrSocketProxy::AssignChannel_DoNotUse(
-    WebrtcProxyChannelWrapper* aWrapper) {
-  mWebrtcProxyChannel = aWrapper;
+void NrSocketProxy::AssignChannel_DoNotUse(WebrtcTCPSocketWrapper* aWrapper) {
+  mWebrtcTCPSocket = aWrapper;
 }
 
 }  // namespace mozilla